client/reader/id-helpers.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External Dependencies
+ */
+
+/**
+ * Internal Dependencies
+ */
+
+export function toValidId( val ) {
+	if ( val === true ) {
+		return undefined;
+	}
```

Comment:
yeah, false going to zero is ok as we later turn 0 into undefined. This is all a bit belt and suspenders as the API only really gives 0 or a valid ID.

I'd be fine with using a more positive case, though I was trying to avoid using a regex. 

If there any way to get a non-finite number out of the Number function?